addons/common/functions/fnc_registerItemReplacement.sqf

```diff
@@ -0,0 +1,114 @@
+ * 1: Item or list of items <STRING or ARRAY>
+ * 2: Replace items that inherit from original item (only if 0 is STRING) <BOOL>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * ["FirstAidKit", "ACE_fieldDressing"] call ace_common_fnc_registerItemReplacement
+ *
+ * Public: Yes
+ */
+params ["_oldItem", "_newItems", ["_replaceInherited", false]];
+
+// CBA player event handler function
+private _fnc_replaceItems = {
+    params ["_unit"];
+
+    private _items = items _unit;
+    if (_items isEqualTo GVAR(oldItems)) exitWith {};
+
+    private _newItems = _items - GVAR(oldItems);
+    if (_newItems isEqualTo []) exitWith {
+        GVAR(oldItems) = _items;
+    };
+
+    _newItems sort true; // Sort so all items of current class can be replaced at once
+    private _replacements = [];
+    private _cfgWeapons = configFile >> "CfgWeapons"; // Microoptimization
+
+    for "_i" from 0 to count _newItems - 1 do {
+        private _item = _newItems#_i;
+
+        // Determine replacement items: direct replacements, ...
+        private _directReplacement = GVAR(itemReplacements) getVariable [_item, []];
+        _replacements append _directReplacement;
+
+        // ... item type replacements ...
+        private _type = getNumber (_cfgWeapons >> _item >> "ItemInfo" >> "type");
+        private _typeReplacements = GVAR(itemReplacements) getVariable ["$" + str _type, []];
+        _replacements append _typeReplacements;
+
+        // ... and inherited replacements
+        {
+            if (_item isKindOf [_x, _cfgWeapons]) then {
+                private _inheritedReplacements = GVAR(itemReplacements) getVariable ["$" + _x, []];
+                _replacements append _inheritedReplacements;
+            };
+        } forEach GVAR(inheritedReplacements);
+
+        // Skip lookup for all following items of this class
+        private _count = 1;
+        while {_newItems#(_i + 1) == _item} do {
+            _count = _count + 1;
+            _i = _i + 1;
+        };
+
+        // Replace all items of current class in list
+        if !(_replacements isEqualTo []) then {
+            _unit removeItems _item;
+
+            for "_j" from 1 to _count do {
+                { [_unit, _x] call FUNC(addToInventory) } forEach _replacements;
+            };
+        };
+    };
+
+    GVAR(oldItems) = items _unit;
+};
+
+// Setup on first run
+if (isNil QGVAR(itemReplacements)) then {
+    GVAR(itemReplacements) = [] call CBA_fnc_createNamespace;
+    GVAR(inheritedReplacements) = [];
+    GVAR(oldItems) = [];
+    ["loadout", _fnc_replaceItems] call CBA_fnc_addPlayerEventHandler;
+};
+
+// Save item replacement
+// $ prefix is used for types (numbers) and replacements with inheritance
+if (_replaceInherited) then {
+    GVAR(inheritedReplacements) pushBack _oldItem;
+    _oldItem = "$" + _oldItem;
+};
+if (_oldItem isEqualType 0) then {
+    _oldItem = "$" + str _oldItem;
+};
+if (_newItems isEqualType "") then {
+    _newItems = [toLower _newItems];
```

Comment:
why `toLower`?

Reply:
Right, that doesn't seem necessary here.